### PR TITLE
New subject viewer: JSONDataViewer

### PR DESCRIPTION
**Pull request by eatyourgreens**
`JSONDataViewer` is a generic JSON subject container, which validates the response from a JSON subject location, then mounts the appropriate viewer component based on the data type (`BarChartViewer`, `LightCurveViewer`, `ScatterPlotViewer` etc.) It replaces individual containers for specific viewers (`BarChartViewerContainer`, `ScatterPlotViewerContainer`, and `ScatterPlotViewerConnector`.) It replaces the static imports for `LightCurveViewer` and `ScatterPlotViewer`, in the classifier's subject viewer, with a single import from `JSONDataViewer` for each of these viewer types.

`JSONData` uses more restrictive models here, with refinements, to match snapshots to types. Bar charts and data series charts cannot have empty data arrays. TESS light curves must have an `x` and a `y` array. Each data type in the `types.union()` **must be uniquely defined by a snapshot**. The `JSONData` tests use real subject URLs, where possible, to test that each snapshot is mapped to the correct model. `JSONData` models used by live projects include:
- `DataSeriesPlot`: SuperWASP: Black Hole Hunters.
- `TESSLightCurve`: Planet Hunters TESS.
- `VariableStarPlots`: Zwicky's Stellar Sleuths.

_Please request review from `@zooniverse/frontend` team or an individual member of that team._ 

## Package
lib-classifier

## How to Review

- staging workflow with JSON data subjects: https://localhost:8080/?project=eatyourgreens/-project-testing-ground&workflow=3762&demo=true
- `JSONDataViewer` in the classifier storybook: http://localhost:6006/?path=/story/subject-viewers-jsondataviewer--data-series&globals=locale:en;locales.en:English;locales.test:Test+Language

The storybook currently has two examples, using two different JSON formats: a SuperWASP subject (`ScatterPlotViewer`) and a PH-TESS subject (`LightCurveViewer`.)

You can use SuperWASP: Black Hole Hunters to check the scatter plot viewer in a live project.
https://local.zooniverse.org:3000/projects/hughdickinson/superwasp-black-hole-hunters/classify/workflow/17386?demo=true

You can use Planet Hunters TESS to check the light curve viewer in a live project.
https://local.zooniverse.org:3000/projects/nora-dot-eisner/planet-hunters-tess/classify/workflow/11235?demo=true

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [x] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [x] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [x] Can submit a classification
- [x] Can sign-in and sign-out
- [x] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
- [x] The PR creator has listed user actions to use when testing the new feature
- [x] Unit tests are included for the new feature
- [x] A storybook story has been created or updated
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)

### CODE REVIEW

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/JSONDataViewer/JSONDataViewer.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import zooTheme from '@zooniverse/grommet-theme'
+import { Box } from 'grommet'
+import { observer } from 'mobx-react'
+import { darken } from 'polished'
```
Comment (goplayoutside3) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/JSONDataViewer/JSONDataViewer.js:4]:
```suggestion
import { Box } from 'grommet'
import { observer } from 'mobx-react'
```

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/JSONDataViewer/JSONDataViewer.stories.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import { Box } from 'grommet'
+import { Provider } from 'mobx-react'
+import { Factory } from 'rosie'
+
+import SubjectViewerStore from '@store/SubjectViewerStore'
```
Comment (goplayoutside3) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/JSONDataViewer/JSONDataViewer.stories.js:5]:
```suggestion
```

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewerContainer.js [hunk 3]
```diff
@@ -62,10 +62,10 @@ const SUBJECT = {
...
 
 export function LightCurveViewerContainer({
+  data = null,
   drawFeedbackBrushes = DEFAULT_HANDLER,
   feedback = false,
   onKeyDown = DEFAULT_HANDLER,
-  subject = SUBJECT,
   loadingState = asyncStates.initialized,
   onError = DEFAULT_HANDLER,
   onReady = DEFAULT_HANDLER
```
Comment (goplayoutside3) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewerContainer.js:71]:
```suggestion
```

These are unused. Should they be used like other subject viewers?

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/ScatterPlotViewer/ScatterPlotViewer.stories.js [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import zooTheme from '@zooniverse/grommet-theme'
 import { Box } from 'grommet'
 import { darken } from 'polished'
-import ScatterPlotViewer from './ScatterPlotViewer'
-import ScatterPlotViewerConnector from './ScatterPlotViewerConnector'
+import ScatterPlotViewer from './'
+import JSONDataViewer from '../JSONDataViewer'
 import { Provider } from 'mobx-react'
 
 import SubjectViewerStore from '@store/SubjectViewerStore'
```
Comment (goplayoutside3) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/ScatterPlotViewer/ScatterPlotViewer.stories.js:8]:
```suggestion
```

File: packages/lib-classifier/src/helpers/subjectViewers/subjectViewers.js [hunk 5]
```diff
@@ -10,6 +10,11 @@ Object.defineProperty(subjectViewers, 'flipbook', {
   enumerable: true
 })
 
+Object.defineProperty(subjectViewers, 'jsonData', {
+  value: 'jsonData',
```
Comment (goplayoutside3) [inline packages/lib-classifier/src/helpers/subjectViewers/subjectViewers.js:14]: Do we also want to add 'jsonData' to `get viewerType()` in the WorkflowConfiguration store action too?

**Review by goplayoutside3 — APPROVED** [review]
Looks good! Just a few non-blocking suggestions/questions.

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewerContainer.js (hunk 3 above)
Comment (eatyourgreens) [reply packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewerContainer.js:71]: I'm not sure. Only @shaunanoordin really understands the LCV code.

File: packages/lib-classifier/src/helpers/subjectViewers/subjectViewers.js (hunk 5 above)
Comment (eatyourgreens) [reply packages/lib-classifier/src/helpers/subjectViewers/subjectViewers.js:14]: Good point. I think we might.

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewerContainer.js (hunk 3 above)
Comment (eatyourgreens) [reply packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewerContainer.js:71]: Looks safe to get rid of them, to me.

**Pull request merged**